frontend/src/views/Lobby.tsx

```diff
@@ -46,15 +52,10 @@ function LobbyPage() {
    * This method uses useCallback so it is not re-built in
    * the useEffect function.
    */
-  const connectUserToRoom = useCallback((roomId: string, nicknameParam: string) => {
+  const connectUserToRoom = useCallback((roomId: string) => {
     connect(roomId).then(() => {
       subscribe(routes(roomId).subscribe, subscribeCallback).then(() => {
-        try {
-          addUser(nicknameParam);
-          setSocketConnected(true);
-        } catch (err) {
-          setError(err.message);
-        }
+        setSocketConnected(true);
```

Comment:
Doesn't this always redirect the user? Is that intended?

It always seems to redirect for me, even when I provide a correct room ID.

Reply:
Yeah if you navigate directly to `/game/lobby` it should always redirect you to the join page unless you already reached the page from the create/join pages (the current code looks only at the user/room location state rather than the roomId query param). My thinking behind this being that users need to enter a nickname then call the PUT endpoint to actually join the room, which currently all happens from the join page. Same thing as above tho - can easily change this if we don't like this design

Reply:
No, this makes sense -- thanks for the explanation!